File: src/main/java/io/github/trainb0y/fabrizoom/config/modmenu/ModMenuIntegration.java
package io.github.trainb0y.fabrizoom.config.modmenu;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import io.github.trainb0y.fabrizoom.config.ConfigScreen;
import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;

@Environment(EnvType.CLIENT)
public class ModMenuIntegration implements ModMenuApi {
	@Override
	public ConfigScreenFactory<?> getModConfigScreenFactory() {
		return ConfigScreen::getConfigScreen;
	}
}

File: src/main/kotlin/io/github/trainb0y/fabrizoom/config/modmenu/ModMenuIntegration.kt
package io.github.trainb0y.fabrizoom.config.modmenu

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import io.github.trainb0y.fabrizoom.config.ConfigScreen
import net.minecraft.client.gui.screen.Screen

class ModMenuIntegration : ModMenuApi {
	override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
		return ConfigScreenFactory<Screen> { parent: Screen? -> ConfigScreen(parent) }
	}
}